src/dataflow/src/source/s3.rs

```diff
@@ -742,103 +702,98 @@ async fn download_object(
     key: &str,
     compression: Compression,
     source_id: &str,
-    skip_bytes: usize,
 ) -> (DownloadStatus, Option<DownloadMetricUpdate>) {
-    let obj = match client
-        .get_object(GetObjectRequest {
+    let reader = RetryReader::new(|state, offset| async move {
+        let range = if offset == 0 {
+            None
+        } else {
+            log::debug!(
+                "Failed to download object: {}/{} attempt={} read={}",
```

Comment:
This message is no longer printed after I think @quodlibetor 's previous fixes. At git revision ``` c2e3a80e1d90188321828ffe381ecc8d79d53289``` , the log from the s3-resumption test contained entries such as

```
2021-09-20T07:37:08.533087Z  WARN dataflow::source::s3: unable to read from SQS queue testdrive-test-2048: Error obtaining body: Error obtaining chunk: error reading a body from connection: end of file before message length reached (9 retries remaining)    
2021-09-20T07:37:08.535367Z DEBUG dataflow::source::s3: source_id=u7/53 testdrive-test-2048/s3/a download_status=Retry { bytes_read: 1208, err: Read(Custom { kind: Other, error: "Error obtaining chunk: error reading a body from connection: end of file before message length reached" }) } 
```

This is no longer the case, even on `main`. toxiproxy reports a bunch of TCP connections interrupted, yet the Mz log is missing any such messages , it is all rainbows and unicorns.

Reply:
Nevermind, I see this error is now printed. So false alarm here.